test/puppetlabs/puppetdb/http/facts_test.clj

```diff
@@ -749,6 +750,26 @@
   ([endpoint query paging-options]
    (:results (raw-query-endpoint endpoint query paging-options))))
 
+(deftestseq large-integer-storage
+  [[version endpoint] facts-endpoints
+   method [:get :post]]
+
+  (let [f {:certname "a.local" :name "large-int" :environment "DEV"
+           :value 3546565741691827463891754698375639827814623789421387401348}]
```

Comment:
I wonder if we should test it closer to the maximum, and then test it when it overflows, make sure it returns something meaningful. Really we want to shake up the rest of the stack here also: Java, JSON libraries etc. BigInteger should handle it though.